amethyst_network/src/simulation/transport/udp.rs

```diff
@@ -0,0 +1,134 @@
+//! Network systems implementation backed by the UDP network protocol.
+use crate::simulation::transport::run_network_recv_system;
+use crate::simulation::{
+    events::NetworkSimulationEvent,
+    requirements::DeliveryRequirement,
+    resource::NetworkSimulationResource,
+    timing::{NetworkSimulationTime, NetworkSimulationTimeSystem},
+    transport::{
+        run_network_send_system, socket::Socket, NETWORK_RECV_SYSTEM_NAME,
+        NETWORK_SEND_SYSTEM_NAME, NETWORK_SIM_TIME_SYSTEM_NAME,
+    },
+};
+use amethyst_core::{
+    bundle::SystemBundle,
+    ecs::{DispatcherBuilder, Read, System, World, Write},
+    shrev::EventChannel,
+};
+use amethyst_error::Error;
+use bytes::Bytes;
+use log::error;
+use std::{
+    io,
+    net::UdpSocket,
+    ops::{Deref, DerefMut},
+};
+
+/// Use this network bundle to add the various underlying UDP network systems to your game.
+pub struct UdpNetworkBundle {
+    recv_buffer_size_bytes: usize,
+}
+
+impl UdpNetworkBundle {
+    pub fn new(recv_buffer_size_bytes: usize) -> Self {
+        Self {
+            recv_buffer_size_bytes,
+        }
+    }
+}
+
+impl<'a, 'b> SystemBundle<'a, 'b> for UdpNetworkBundle {
+    fn build(
+        self,
+        _world: &mut World,
+        builder: &mut DispatcherBuilder<'_, '_>,
+    ) -> Result<(), Error> {
+        builder.add(
```

Comment:
we could use method chaining here

Reply:
Method chaining? `DispatcherBuilder::add` doesn't return a value.

Reply:
Ah there's a `with` method that allows for that. I'll use that.

Reply:
Ah actually I can't do that because it requires that I own the builder but I only have a mutable reference.

Reply:
Ahh, oke!